Modules/_cursesmodule.c

```diff
@@ -170,34 +170,72 @@ class _curses.window "PyCursesWindowObject *" "&PyCursesWindow_Type"
+static int CURSES_INITSCR_CALLED = FALSE;
 
 /* Tells whether start_color() has been called to initialise color usage. */
-static int initialisedcolors = FALSE;
+static int CURSES_INITIALISED_COLORS = FALSE;
 
-static char *screen_encoding = NULL;
+static char *CURSES_SCREEN_ENCODING = NULL;
 
 /* Utility Macros */
-#define PyCursesSetupTermCalled                                         \
-    if (initialised_setupterm != TRUE) {                                \
-        PyErr_SetString(PyCursesError,                                  \
-                        "must call (at least) setupterm() first");      \
-        return 0; }
-
-#define PyCursesInitialised                             \
-    if (initialised != TRUE) {                          \
-        PyErr_SetString(PyCursesError,                  \
-                        "must call initscr() first");   \
-        return 0; }
-
-#define PyCursesInitialisedColor                                \
-    if (initialisedcolors != TRUE) {                            \
-        PyErr_SetString(PyCursesError,                          \
-                        "must call start_color() first");       \
-        return 0; }
+
+/*
+ * Check that GLOBAL_FLAG is set, or advise to call FUNC_TO_CALL().
+ *
+ * This macro returns 0 on error so that it can be used both in a
+ * boolean context and in a function that returns a (PyObject *).
+ */
+#define CHECK_STATE_FLAG(FUNC_TO_CALL, GLOBAL_FLAG)                 \
+    do {                                                            \
+        if ((GLOBAL_FLAG) != TRUE) {                                \
+            assert(PyCursesError != NULL);                          \
+            PyErr_SetString(PyCursesError,                          \
+                            "must call " FUNC_TO_CALL "() first");  \
+            return 0;                                               \
+        }                                                           \
+    } while (0)
+
+#define PyCursesSetupTermCalled \
+    CHECK_STATE_FLAG("setupterm", CURSES_SETUPTERM_CALLED)
+
+#define PyCursesInitialised \
+    CHECK_STATE_FLAG("initscr", CURSES_INITSCR_CALLED)
+
+#define PyCursesInitialisedColor \
+    CHECK_STATE_FLAG("start_color", CURSES_INITIALISED_COLORS)
+
+#define CHECK_NOT_NULL_OR_ERROR(VALUE)  \
+    do {                                \
+        if ((VALUE) == NULL) {          \
+            goto error;                 \
+        }                               \
+    } while (0)
+
+#define CHECK_RET_CODE_OR_ERROR(STATUS) \
+    do {                                \
+        if ((STATUS) < 0) {             \
+            goto error;                 \
+        }                               \
+    } while (0)
+
+#define CHECK_RET_FLAG_OR_ERROR(STATUS) \
+    do {                                \
+        if (!(STATUS)) {                \
+            goto error;                 \
+        }                               \
+    } while (0)
```

Comment:
I think it's a matter of personal taste but I can also understand your stand on that matter. Personally, what I find hard to read is when you have all those `if(...) { goto error; }` spread on 3 lines. I would find them much easier to read if PEP 7 allowed me to write `if (!(STATUS)) goto error;`. On the other hand, with those macros, you may be also introduce errors where you need to free resources before jumping to `error` so they should be used carefully...

I had alternatives where you would just put whatever cleanup statements you want when invoking the macro but I felt it would be even less readable...